### ISSUE TITLE
Proxy settings

### ISSUE THREAD

**Issue by figue** [graysky2/clean-chroot-manager#61]
Hello, I need to include proxy variables (http_proxy, https_proxy, etc) inside chroot in order to connect to internet. Is there a way to achieve it? Tried to place in different profile files (~/.config/clean-chroot-manager.conf, root/etc/profile, myusername/etc/profile etc) but no luck... pacman can't connect to internet.

Thank you!

**Comment by graysky2** [comment]
Never used a proxy ... if you can get it working with the standard build scripts, it would likely be easy to adapt ccm.  Have you consulted google and or the forums?

**Comment by figue** [comment]
Yes, I tried several things. The initial setup of chroot works (ccm64 c), but when makepkg is launched doesn't.

In summay, in this case, the shell that launch makepkg have to have these variables:

FTP_PROXY=http://127.0.0.1:8889
HTTPS_PROXY=http://127.0.0.1:8889
HTTP_PROXY=http://127.0.0.1:8889
NO_PROXY=127.0.0.1,localhost
RSYNC_PROXY=http://127.0.0.1:8889
ftp_proxy=http://127.0.0.1:8889
http_proxy=http://127.0.0.1:8889
https_proxy=http://127.0.0.1:8889
no_proxy=127.0.0.1,localhost
rsync_proxy=http://127.0.0.1:8889

I have them in my sudoers file:

Defaults env_keep += "PROXY https_proxy http_proxy ftp_proxy rsync_proxy no_proxy HTTP_PROXY HTTPS_PROXY FTP_PROXY RSYNC_PROXY NO_PROXY"

So one solution could be allow to bypass a custom variable to the makepkg shell (useful maybe for other purposes), hardcode these variables into ccm64 or allow to install a package into chroot before build (wget in this case to force proxy there).

**Comment by graysky2** [comment]
I don't have a proxy setup in order to properly test.  Is there some documentation for using devtools behind a proxy?

**Comment by daftaupe** [comment]
> I don't have a proxy setup in order to properly test. Is there some documentation for using devtools behind a proxy?

No, that's actually one problematic thing with devtools for people using (or being forced to use) a proxy.

**Comment by graysky2** [comment]
I am going to close since setting up proxy services is beyond the scope of this tool.
